feat: upgrade to 25.11 and re-use upstream's package expressions

ConnorBaker opened the pull request:
###### Description of changes

> [!NOTE]
>
> This PR updates the Nixpkgs flake input from `nixos-25.05` to `nixos-25.11`.

Since the introduction of [nixos-cuda/cuda-legacy](https://github.com/nixos-cuda/cuda-legacy) and the various changes I've made upstream to support it, we can re-use many, many more package expressions from upstream, meaning we no longer need to vendor them.

This PR essentially removes the vendored package expressions introduced as part of JetPack 6/CUDA 12 support because upstream was pending a refactor (the package expressions were largely taken from my WIP refactor branch for upstream).

As an additional change, all CUDA package sets are now created as *extensions* of the CUDA package sets available upstream (see https://github.com/NixOS/nixpkgs/blob/c5220bcad8d7e599432db935f1159a81d5019dd4/doc/languages-frameworks/cuda.section.md?plain=1#L66), meaning that we benefit from using the same setup hooks and gain the same niceties they do (e.g., `backendStdenv`).

Additionally, the NixOS module setting up graphics for Jetson devices includes many more libraries, matching the larger set of functionality included in NVIDIA's desktop drivers.

Lastly, specific changes for CUDA 11.4/JetPack 5:

- we are able to use upstream's package expressions modulo a little bit of normalization of unpacked debian installers
- libnvinfer-samples has been removed in favor of cudaPackages.tensorrt-samples
- TensorRT 8.5.2 is provided for x86_64-linux to match the version provided for Jetson

###### Testing

- [x] Verify `samples.combinded-test` works across JetPack 5/6/7
- [x] Internal HITL testing
  - [x] Xavier JP5
  - [x] Orin JP5
  - [x] Orin JP6
  - [x] Thor JP7

ConnorBaker committed:
tree-wide: re-use as much of upstream as possible

I've updated the package expressions for CUDA-related software in Nixpkgs to support as many releases as possible to support nixos-cuda/cuda-legacy.

This allows us to re-use their package expressions, even if we have to make small tweaks to them.

As a result, for JetPack 6/CUDA 12 and later, we can use upstream's packaging.

For JetPack 5/CUDA 11, we still need to use the debian installers. Thankfully, content-wise they're similar enough to the binary archives NVIDIA releases that we can just re-organize their contents and pass them along to upstream's packaging logic.

Signed-off-by: Connor Baker <ConnorBaker01@gmail.com>
Signed-off-by: Connor Baker <cbaker2@anduril.com>

elliotberman commented on overlay.nix:
```diff
@@ -124,48 +127,57 @@ in
-      final.cudaPackages_12_6
-    else
-      prev.cudaPackages_12;
+  # Override upstream's manifest selection so the version of TensorRT used is consistent
```
```suggestion
  # Override upstream's manifest selection so the version of TensorRT used is consistent with L4T
```

ConnorBaker committed:
pkgs/samples: use cudaPackages.backendStdenv where appropriate

Signed-off-by: Connor Baker <cbaker2@anduril.com>

ConnorBaker committed:
apply overlay at most once

Signed-off-by: Connor Baker <cbaker2@anduril.com>

ConnorBaker committed:
ensure aliases always refer to final to reflect changes made to intermediates

Signed-off-by: Connor Baker <ConnorBaker01@gmail.com>

ConnorBaker committed:
vendor TensorRT for CUDA 11.4 for x86_64-linux

Signed-off-by: Connor Baker <ConnorBaker01@gmail.com>

ConnorBaker committed:
vendor cuda_profiler_api for CUDA 11.4 for x86_64-linux

Signed-off-by: Connor Baker <ConnorBaker01@gmail.com>

ConnorBaker committed:
l4t-3d-core: remove unused ld.so.conf and doc

Signed-off-by: Connor Baker <ConnorBaker01@gmail.com>

ConnorBaker committed:
hardware.graphics.extraPackages: include most driver libs

A number of libraries needed for TensorRT were missing, causing inference to fail.

This is directly a result of re-using upstream's packaging expressions, but is not a bad thing -- the libraries needed are provided by the system and we shouldn't be patchelf-ing them in during the package build, because in the general case we do not have access to system libraries at runtime. They should instead be discovered through LD_LIBRARY_PATH.

Signed-off-by: Connor Baker <ConnorBaker01@gmail.com>

ConnorBaker committed:
fix flake check and format

Signed-off-by: Connor Baker <ConnorBaker01@gmail.com>

ConnorBaker committed:
support cudaPackages.tensorrt-samples for JetPack 5/CUDA 11.4

Signed-off-by: Connor Baker <ConnorBaker01@gmail.com>

ConnorBaker committed:
flake.nix: only enable Jetson capabilities for legacyPackages.aarch64-linux

Signed-off-by: Connor Baker <ConnorBaker01@gmail.com>

ConnorBaker commented on pkgs/l4t/l4t-3d-core.nix:
```diff
@@ -50,6 +50,9 @@ buildFromDebs {
       fi
     done
+
+    rm lib/ld.so.conf
```
Good catch, fixed in latest (force-pushed).

ConnorBaker committed:
unify buildFromDebs with normalizeDebs

Adds some missing normalization passes to normalizeDebs and introduces pre/postDebNormalization hooks to handle odd packages.

Signed-off-by: Connor Baker <cbaker2@anduril.com>
